src/services/insertBaseRoute.ts

```diff
@@ -0,0 +1,19 @@
+import { combinePath } from '@/services/combinePath'
+import { path } from '@/services/path'
+import { Routes } from '@/types'
+import { isNestedArray, stringHasValue } from '@/utilities/guards'
+
+export function insertBaseRoute(routesOrArrayOfRoutes: Routes | Routes[], base?: string): Routes {
+  const routes = isNestedArray(routesOrArrayOfRoutes) ? routesOrArrayOfRoutes.flat() : routesOrArrayOfRoutes
```

Comment:
I'd probably pull this out so that this method isn't also responsible for flattening the routes.